tests/test_kv_store.c

```diff
@@ -230,6 +231,48 @@ static void test_kvstore_iterate(void **state)
 	release_test_data(test_iov);
 }
 
+
+static void test_kvstore_merge_op(void **state)
+{
+	struct iovec              test_iov[MAX_TEST_ENTRIES];
+	size_t                    data_size;
+	kv_store_value_flags_t    flags;
+	sid_resource_t *          kv_store_res = NULL;
+	struct test_kv_update_arg update_arg;
+	size_t                    test_data_size, kv_data_size;
+	size_t                    index;
+	struct kv_store_value *   kv_return_value;
+
+	kv_store_res = sid_resource_create(SID_RESOURCE_NO_PARENT,
+	                                   &sid_resource_type_kv_store,
+	                                   SID_RESOURCE_RESTRICT_WALK_UP,
+	                                   "testkvstore",
+	                                   &test_kv_store_res_params,
+	                                   SID_RESOURCE_PRIO_NORMAL,
+	                                   SID_RESOURCE_NO_SERVICE_LINKS);
+
+	test_data_size = add_test_data(test_iov);
+
+	assert_ptr_not_equal(kv_store_set_value(kv_store_res,
+	                                        MERGE_KEY,
+	                                        &test_iov,
+	                                        MAX_TEST_ENTRIES,
+	                                        KV_STORE_VALUE_VECTOR,
+	                                        KV_STORE_VALUE_OP_MERGE,
+	                                        _kv_write_always,
+	                                        &update_arg),
+	                     NULL);
+	assert_int_equal(kv_store_num_etries(kv_store_res), 1);
+	kv_return_value = kv_store_get_value(kv_store_res, MERGE_KEY, &kv_data_size, NULL);
+
+	assert_int_equal(test_data_size, kv_data_size);
+	assert_int_equal(kv_return_value->size, kv_data_size);
+	assert_int_equal(memcmp(kv_return_value->data, test_iov, kv_data_size), 0);
```

Comment:
@prajnoha the test is failing on the memcmp().  when I step through the hash lookup all looks fine.  I think it is related to not using a KV_STORE_VALUE_REF | KV_STORE_VALUE_VECTOR parameter.  Is it obvious to you what I'm doing wrong?